api/src/main/scala/dcos/metronome/api/v1/controllers/JobScheduleController.scala

```diff
@@ -76,12 +76,15 @@ class JobScheduleController(
     }
 
     withJobSpec(id) { spec =>
-      jobSpecService.updateJobSpec(id, deleteSchedule)
-        .map(_ => Ok)
-        .recover{
-          case JobSpecDoesNotExist(_)       => NotFound(UnknownJob(id))
```

Comment:
The impl of this controller is pretty weird: `withJobSpec` reads the jobSpec, and will result in a `JobSpecDoesNotExist` exception if it doesn't exist. In case we could fetch the jobSpec, we'll call `updateJobSpec`, which will basically do the same – try to fetch the spec, apply a function or throw `JobSpecDoesNotExist`.

The impl could simply _try_ to create/update/delete the spec and handle the exceptions accordingly. There's quite some DRY's in here that _could_ be resolved.